### PR TITLE
chore: refactor code to abstract out tupleIDInSortedSet

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Refactor code to abstract out tupleIDInSortedSet so that it can be unit tested in its entirity.

## References
Follow up on https://github.com/openfga/openfga/pull/1833

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: internal/graph/check.go [hunk 1]
```diff
@@ -667,6 +667,25 @@ func (c *LocalChecker) hasCycle(req *ResolveCheckRequest) bool {
...
+			return false, err
+		}
+
+		_, objectID := tuple.SplitObject(t.GetObject())
+		if objectIDs.Exists(objectID) {
+			return true, nil
+		}
+	}
+	return false, nil
+}
```
Comment (sergiught) [inline internal/graph/check.go:687]: **(suggestion):** We could further reduce complexity here and improve on the description. 

```suggestion
// tupleIDInSortedSet checks if any tuple object ID returned by the 
// iterator exists in the provided sorted set of object IDs.
func tupleIDInSortedSet(
	ctx context.Context,
	filteredIter *storage.ConditionsFilteredTupleKeyIterator,
	objectIDs storage.SortedSet,
) (bool, error) {
	for {
		t, err := filteredIter.Next(ctx)
		if errors.Is(err, storage.ErrIteratorDone) {
			return false, nil
		}
		if err != nil {
			return false, err
		}

		_, objectID := tuple.SplitObject(t.GetObject())
		if objectIDs.Exists(objectID) {
			return true, nil
		}
	}
}
```

File: internal/graph/check.go [hunk 2]
```diff
@@ -731,24 +744,21 @@ func (c *LocalChecker) buildCheckAssociatedObjects(req *ResolveCheckRequest, obj
...
+		if allowed {
+			span.SetAttributes(attribute.Bool("allowed", true))
 		}
-		return response, nil
+		return &ResolveCheckResponse{
+			Allowed: allowed,
+			ResolutionMetadata: &ResolveCheckResponseMetadata{
+				DatastoreQueryCount: reqCount,
+			},
+		}, nil
```
Comment (sergiught) [inline internal/graph/check.go:761]: **(nitpick):** Keeping var scope as small as possible.

```go

allowed, err := tupleIDInSortedSet(ctx, filteredIter, objectIDs)
if err != nil {
	telemetry.TraceError(span, err)
	return nil, err
}

if allowed {
	span.SetAttributes(attribute.Bool("allowed", true))
}

reqCount := req.GetRequestMetadata().DatastoreQueryCount + 1

return &ResolveCheckResponse{
	Allowed: allowed,
	ResolutionMetadata: &ResolveCheckResponseMetadata{
		DatastoreQueryCount: reqCount,
	},
}, nil

```

File: internal/graph/check_test.go [hunk 3]
```diff
@@ -2198,3 +2198,68 @@ func TestCycleDetection(t *testing.T) {
...
+			return true, nil
+		}
+		return false, fmt.Errorf("condition not found")
+	}
+
+	tests := []struct {
+		name          string
+		tuples        []*openfgav1.TupleKey
+		objectIDs     []string
+		expectedError bool
```
Comment (sergiught) [inline internal/graph/check_test.go:2214]: **(suggestion):** Could we change this to a string? And match against the actual error message? Wdyt?

Comment (adriantam) [reply internal/graph/check_test.go:2214]: Similar to https://github.com/openfga/openfga/pull/1840#discussion_r1716956867, the error message is generated from external function.  Hence, we don't want to tie the assertion to whatever error string the external function generate.

**Review by sergiught — APPROVED** [review]

**Review by elbuo8 — APPROVED** [review]

**Pull request merged**